packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2072,6 +2072,14 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         return this.finishNode(nonNullExpression, "TSNonNullExpression");
       }
 
+      if (
+        this.match(tt.questionDot) &&
```

Comment:
We can not consume questionDot after peeking `<` because the token after closing `>` may not form a call expression. In this case we should stop at `?.`:

```ts
f?.<string>[1]
```
should throw.

Reply:
We could also just consume `?.`, but then throw if we don't find `(`. By doing so we don't need to look ahead to check if there is the `(`.

Reply:
I returned a raise if no '(' after parsing type.